config/configunmarshaler/defaultunmarshaler.go

```diff
@@ -269,6 +271,8 @@ func unmarshalService(rawService serviceSettings) (config.Service, error) {
 		ret.Extensions = append(ret.Extensions, id)
 	}
 
+	ret.Gates = featuregate.Apply(rawService.Gates)
```

Comment:
I think the "service" should resolve/apply the gates not the configunmarshaler, this should just "get the user  configuration for the gates".

Reply:
Actually I think I understand what is missing here... a way to provide a "default config" for the "service" section of the configuration... I need to think a bit more here.